airgun/entities/host_new.py

```diff
@@ -272,15 +435,146 @@ def get_parameters(self, entity_name):
         return view.parameters.read()
 
     def add_new_parameter(self, entity_name, parameter_name, parameter_type, parameter_value):
+        """
+        Function that adds new parameter to the host
+
+        Args:
+            entity_name: host on which we want to add parameter
+            parameter_name: Name of the parameter to be added
+            parameter_type: Type of the parameter to be added
+                [string, boolean, integer, real, array, hash, yaml, json]
+            parameter_value: Value of the parameter to be added
+
+        Raises:
+            ValueError: If parameter type is not valid
+            ValueError: If parameter with the same name already exists
+        """
+
+        # Sanitize input
+        parameter_type = parameter_type.lower()
+        available_types = ['string', 'boolean', 'integer', 'real', 'array', 'hash', 'yaml', 'json']
+        if parameter_type not in available_types:
+            raise ValueError(f'Parameter type {parameter_type} is not valid')
+
         view = self.navigate_to(self, 'NewDetails', entity_name=entity_name)
         view.wait_displayed()
         self.browser.plugin.ensure_page_safe()
+        view.parameters.click()
+        view.parameters.searchbar.fill(parameter_name)
+        view.wait_displayed()
+        # Check if parameter with given name does not exist. If it does, raise ValueError.
+        if view.parameters.parameters_table.row_count != 0:
+            if view.parameters.parameters_table[0][0].text == parameter_name:
+                raise ValueError(f'Parameter with name {parameter_name} already exists')
+
         view.parameters.add_parameter.click()
-        view.parameters.new_parameter_name.fill(parameter_name)
-        view.parameters.new_parameter_type.fill(parameter_type)
-        view.parameters.new_parameter_value.fill(parameter_value)
+        view.parameters.parameter_name_input.fill(parameter_name)
+        view.parameters.parameter_type_input.fill(parameter_type)
+        view.parameters.parameter_value_input.fill(parameter_value)
```

Comment:
Same idea as above with calling `fill` for this whole thing

Reply:
Similar explanation as here: https://github.com/SatelliteQE/airgun/pull/833#discussion_r1179050031 
It is not implemented as form... and widgetastic does not support forms.